pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+
+	"github.com/golang/glog"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	awsclient "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/aws/client"
+	ccerrors "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/errors"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/pkg/errors"
+)
+
+const (
+	jumpRole      = "ba"
+	defaultRegion = "us-east-1" // TODO extract to own config
+	credentialDir = "~/.aws"    // TODO extract to own config
+)
+
+const (
+	// the ClusterDeployment label key we're expecting to be able to pull the ClusterID from
+	cdClusterIDLabelKey string = "api.openshift.com/id"
+)
+
+// CloudClient is the client used for handling AWS Cloud Handler requests
+type CloudClient struct {
+	// clusterDeployment is the ClusterDeployment for the target cluster (the one
+	// for which a login url is requested)
+	clusterDeployment *hivev1.ClusterDeployment
+
+	// awsClient is the awsClient that has the BACKPLANE user credentials.  This client
+	// is used to create other clients in order to perform the necessary role chaining to
+	// access a customer's account via STS assume roles.
+	awsClient awsclient.AwsClient
+
+	// ocmClient is a client that can query OCM in order to get the AccountClaim
+	ocmClient ocm.Client
+
+	// awsClientBuilder is responsible for building the aws client in order to
+	// allow mocking during testing
+	awsClientBuilder awsclient.BuilderIface
+
+	// region is the region in which we are building sts clients
+	region string
+}
+
+// NewCloudClient makes a new AWS CloudClient based on the ClusterDeployment and
+// handles initial configuration building for a single request.
+func NewCloudClient(cloudConfig *interface{}, ocmClient ocm.Client, cd *hivev1.ClusterDeployment) (*CloudClient, error) {
```

Comment:
How is the cloudConfig used?

Reply:
It's not used yet, it still uses hardcoded values, like the jumpRole, defaultRegion and credentialPosition. That's basically a placeholder. I mean at the moment it works on my machine.
I wanted to put the configuration injection into an own story. And rather try to test it like it is. Just to split the work a bit

Reply:
You can also create an empty struct instead of an empty interface

Reply:
interesting, but then when not providing the same struct type in the function call it will break no? 
From what I understood, you can pass any value to a parameter that implements an empty interface. 
Any type matches an empty interface

Reply:
Yeah, just I don't like seeing a pointer to an empty Interface